WS Zero variant lacks PIN_LED definition => FreeRTOS compliation issue

Issue is as follows:

`In file included from <User>\.platformio\packages\framework-arduinopico\variants\waveshare_rp2040_zero/pins_arduino.h:66,
                 from <User>\.platformio\packages\framework-arduinopico\cores\rp2040/Arduino.h:30,
                 from <User>\.platformio\packages\framework-arduinopico\libraries\FreeRTOS\src\variantHooks.cpp:35:
<User>\.platformio\packages\framework-arduinopico\libraries\FreeRTOS\src\variantHooks.cpp: In function 'void prvSetMainLedOn()':
<User>\.platformio\packages\framework-arduinopico\variants\generic\common.h:9:21: error: 'PIN_LED' was not declared in this scope
    9 | #define LED_BUILTIN PIN_LED
      |                     ^~~~~~~
<User>\.platformio\packages\framework-arduinopico\libraries\FreeRTOS\src\variantHooks.cpp:219:15: note: in expansion of macro 'LED_BUILTIN'
  219 |     gpio_init(LED_BUILTIN);
      |               ^~~~~~~~~~~
<User>\.platformio\packages\framework-arduinopico\libraries\FreeRTOS\src\variantHooks.cpp: In function 'void prvBlinkMainLed()':
<User>\.platformio\packages\framework-arduinopico\variants\generic\common.h:9:21: error: 'PIN_LED' was not declared in this scope
    9 | #define LED_BUILTIN PIN_LED
      |                     ^~~~~~~
<User>\.platformio\packages\framework-arduinopico\libraries\FreeRTOS\src\variantHooks.cpp:228:14: note: in expansion of macro 'LED_BUILTIN'
  228 |     gpio_put(LED_BUILTIN, !gpio_get(LED_BUILTIN));
      |              ^~~~~~~~~~~
*** [.pio\build\PicoZero\lib16f\FreeRTOS\variantHooks.cpp.o] Error 1`


The main.cpp is as simple as:

`
#include <Arduino.h>
#include <FreeRTOS.h>

void setup()
{
}

void loop()
{
}
`

Compiling rpipicow has no issues, but PicoZero has because pins_arduino.h has no definition of PIN_LED (instead there is a definition of PIN_NEOPIXEL skipping LED_BUILTIN definition; so I suspect a completely different implementation for LED as the reason behind, but still the compilation of FreeRTOS needs LED_BUILTIN.

`// NeoPixel
#define PIN_NEOPIXEL   (16u)
//#define LED_BUILTIN PIN_NEOPIXEL`

As a workaround, one can define PIN_LED in platformio as a build flag:
`
[platformio]
build_cache_dir = .cache
default_envs = rpipicow ,PicoZero
src_dir = src
data_dir = data

[env]
platform = https://github.com/maxgerhardt/platform-raspberrypi.git
framework = arduino
board_build.core = earlephilhower

[env:PicoZero]
board = waveshare_rp2040_zero
build_flags = -DPIN_LED=(16u)

[env:rpipicow]
board = rpipicow
`

--- a/libraries/FreeRTOS/keywords.txt
+++ b/libraries/FreeRTOS/keywords.txt
@@ -2,6 +2,8 @@
 # https://arduino.github.io/arduino-cli/library-specification/#keywords
 # Formatted by a single true tab (not spaces)
 
+FreeRTOS	KEYWORD1
+
 # Datatypes (KEYWORD1)
 StackType_t	KEYWORD1
 BaseType_t	KEYWORD1

--- a/libraries/FreeRTOS/src/variantHooks.cpp
+++ b/libraries/FreeRTOS/src/variantHooks.cpp
@@ -216,16 +216,20 @@ void vApplicationTickHook(void) {
     Private function to enable board led to use it in application hooks
 */
 void prvSetMainLedOn(void) {
+#ifdef LED_BUILTIN
     gpio_init(LED_BUILTIN);
     gpio_set_dir(LED_BUILTIN, true);
     gpio_put(LED_BUILTIN, true);
+#endif
 }
 
 /**
     Private function to blink board led to use it in application hooks
 */
 void prvBlinkMainLed(void) {
+#ifdef LED_BUILTIN
     gpio_put(LED_BUILTIN, !gpio_get(LED_BUILTIN));
+#endif
 }
 
 #endif

--- a/variants/generic/common.h
+++ b/variants/generic/common.h
@@ -6,7 +6,9 @@
 #define NUM_ANALOG_OUTPUTS  (0u)
 #define ADC_RESOLUTION      (12u)
 
+#ifdef PIN_LED
 #define LED_BUILTIN PIN_LED
+#endif
 
 static const uint8_t D0 = (0u);
 static const uint8_t D1 = (1u);

